syntax = "proto2";

package dota;

message additional_unit {
	required string unitname = 1;
	required uint32 item_0 = 2;
	required uint32 item_1 = 3;
	required uint32 item_2 = 4;
	required uint32 item_3 = 5;
	required uint32 item_4 = 6;
	required uint32 item_5 = 7;
	required uint32 backpack_0 = 8;
	required uint32 backpack_1 = 9;
	required uint32 backpack_2 = 10;
}

message ability {
	required uint32 ability = 1;
	required uint32 time = 2;
	required uint32 level = 3;
}

message player {
	required uint64 account_id = 1;
	required uint32 player_slot = 2;
	required uint32 hero_id = 3;
	required uint32 item_0 = 4;
	required uint32 item_1 = 5;
	required uint32 item_2 = 6;
	required uint32 item_3 = 7;
	required uint32 item_4 = 8;
	required uint32 item_5 = 9;
	required uint32 backpack_0 = 10;
	required uint32 backpack_1 = 11;
	required uint32 backpack_2 = 12;
	required uint32 kills = 13;
	required uint32 deaths = 14;
	required uint32 assists = 15;
	required uint32 leaver_status = 16;
	required uint32 last_hits = 17;
	required uint32 denies = 18;
	required uint32 gold_per_min = 19;
	required uint32 xp_per_min = 20;
	required uint32 level = 21;
	required uint32 hero_damage = 22;
	required uint32 tower_damage = 23;
	required uint32 hero_healing = 24;
	required uint32 gold = 25;
	required uint32 gold_spent = 26;
	required uint32 scaled_hero_damage = 27;
	required uint32 scaled_tower_damage = 28;
	required uint32 scaled_hero_healing = 29;
	repeated ability ability_upgrades = 30;
	repeated additional_unit additional_units = 31;
}

message players {
	repeated player players = 1;
}

